.github/workflows/update-xla-commit-hash.yml

```diff
@@ -0,0 +1,62 @@
+name: Update xla commit hash
+
+on:
+  schedule:
+    - cron: 0 0 * * *
```

Comment:
Should we make this not start at the 0th minute at the 0th hour as I assume lots of workflows do this.

Also include comment the cron "every night at ___"